src/main/java/com/google/sps/servlets/getGamesServlet.java

```diff
@@ -0,0 +1,149 @@
+  private static final Logger logger = Logger.getLogger(getGamesServlet.class.getName());
+
+  // This method is used to access the api key stored in gcloud secret manager.
+  public String accessSecretVersion(String projectId, String secretId, String versionId) throws IOException {
+    try (SecretManagerServiceClient client = SecretManagerServiceClient.create()){
+      SecretVersionName secretVersionName = SecretVersionName.of(projectId, secretId, versionId);
+
+      // Access the secret version.
+      AccessSecretVersionResponse response = client.accessSecretVersion(secretVersionName);
+
+      // Return the secret payload as a string.
+      return response.getPayload().getData().toStringUtf8();
+
+      // If running on local server return this line instead: return System.getenv("airtable_api_key");
+    }
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    StringBuilder strBuf = new StringBuilder();  
+    HttpURLConnection conn = null;        
+    BufferedReader reader = null;
+
+    try {
+      // Get hidden api key from gcloud secret manager.
+      String airtableapiKey = accessSecretVersion("298755462", "airtable_api_key", "1");
+      URL url = new URL(baseURL + airtableapiKey);
+      conn = (HttpURLConnection) url.openConnection();
+
+      conn.setRequestMethod("GET");
+      conn.setRequestProperty("Accept", "application/json");
+
+      if (conn.getResponseCode() != HttpURLConnection.HTTP_OK) {
+        throw new RuntimeException("HTTP GET Request Failed with Error code : "
+          + conn.getResponseCode());
+      }
+      
+      // Using IO Stream with Buffer to increase efficiency.
+      reader = new BufferedReader(new InputStreamReader(conn.getInputStream(), "utf-8"));
+      String output = null;
+
+      while ((output = reader.readLine()) != null) {
+        strBuf.append(output);
+      }
+    } catch (MalformedURLException e) {
+        response.setContentType("text/html");
+        response.getWriter().println("URL is not correctly formatted");
+        return;        
+    } catch (IOException e) {
+        response.setContentType("text/html");
+        response.getWriter().println("Cannot retrieve information from provided URL");
+        return;
+    } finally {
+        if (reader != null) {
+          try {
+            reader.close();
+          } catch (IOException e) {
+            logger.log(Level.WARNING, e.getMessage());
+          }
+        }
+        if (conn != null) {
+          conn.disconnect();
+        }
+    }
+
+    String games = strBuf.toString();
+
+    // Formats data to be fetched in js.
+    JSONObject obj = new JSONObject(games);
+    JSONArray gamesData = obj.getJSONArray("records");
+    int numGames = gamesData.length();
+
+    List<Game> gamesArray = new ArrayList<Game>();
+
+    for (int i = 0; i < numGames; ++i) {
```

Comment:
I believe this would make the value of i in the first loop iteration 1. Is this intended?

Reply:
yes!

Reply:
Ok cool

Reply:
I think in for loops i++ and ++i are identical. The difference between i++ and ++i is the return value, but we don't use the return value in for loop.